ToolBox/KrakenDataDownloader/KrakenDataDownloader.cs

```diff
@@ -0,0 +1,141 @@
+            var endUnixTime = ToUnixTime(endUtc) * 1000000000;
+            var url = string.Format(UrlPrototype, symbol.Value, startUnixTime);
+            List<List<string>> data;
+
+            using (var client = new WebClient())
+            {
+                var response = client.DownloadString(url);
+                dynamic result = JsonConvert.DeserializeObject<dynamic>(response);
+                if (result.error.Count != 0)
+                {
+                    throw new Exception("Error in Kraken API: " + result.error[0]);
+                }
+
+                data = result.result[symbol.Value].ToObject<List<List<string>>>();
+
+                var last = Convert.ToInt64(result.result.last);
+                while (last < endUnixTime)
+                {
+                    url = string.Format(UrlPrototype, symbol.Value, last);
+                    response = client.DownloadString(url);
+                    result = JsonConvert.DeserializeObject<dynamic>(response);
+
+                    var errorCount = 0;
+                    while (result.error.Count != 0 && errorCount < 10)
+                    {
+                        errorCount++;
+                        System.Threading.Thread.Sleep(6000);
+                        response = client.DownloadString(url);
+                        result = JsonConvert.DeserializeObject<dynamic>(response);
+                    }
+
+                    if (result.error.Count != 0 && errorCount >= 10)
+                    {
+                        throw new Exception("Error in Kraken API: " + result.error[0]);
+                    }
+
+                    List<List<string>> newData = result.result[symbol.Value].ToObject<List<List<string>>>();
+                    data.AddRange(newData);
+                    last = Convert.ToInt64(result.result.last);
+                }
+            }
+
+            foreach (var i in data)
+            {
+                var time = FromUnixTime(Convert.ToInt64(i[2].Split('.')[0]));
+                if (time > endUtc)
+                {
+                    break;
+                }
+
+                var value = Decimal.Parse(i[0]);
+
+                yield return new Tick
+                {
+                    Time = time,
+                    Symbol = symbol,
+                    Value = value,
+                    AskPrice = value,
+                    BidPrice = value,
+                    TickType = TickType.Trade
+                };
+            }
+        }
+
+        /// <summary>
+        /// Convert a DateTime object into a Unix time long value
+        /// </summary>
+        /// <param name="utcDateTime">The DateTime object (UTC)</param>
+        /// <returns>A Unix long time value.</returns>
+        /// <remarks>When we move to NET 4.6, we can replace this with DateTimeOffset.ToUnixTimeSeconds()</remarks>
+        private static long ToUnixTime(DateTime utcDateTime)
```

Comment:
Please use similar methods in the Time class

Reply:
I think DateTimeOffset.ToUnixTimeSeconds() is only available in .NET 4.6. It is not allowing me to use in in the VS project.  It seems we are using a version of .NET 4.5. Is this correct?

Reply:
`QuantConnect.Time.DateTimeToUnixTimeStamp` and `QuantConnect.Time.UnixMillisecondTimeStampToDateTime`